rotkehlchen/db/custom_assets.py

```diff
@@ -0,0 +1,111 @@
+        Adds a custom new custom asset into `assets` and `custom_assets` table.
+        May raise:
+        - InputError if the combination of the name & type / asset with identifier already exist.
+        """
+        self.check_if_custom_asset_exists(custom_asset)
+        with GlobalDBHandler().conn.write_ctx() as global_db_write_cursor, self.db.user_write() as db_write_cursor:  # noqa: E501
+            global_db_write_cursor.execute(
+                'INSERT INTO assets(identifier, name, type) VALUES (?, ?, ?)',  # noqa: E501
+                (
+                    custom_asset.identifier,
+                    custom_asset.name,
+                    AssetType.CUSTOM_ASSET.serialize_for_db(),
+                ),
+            )
+            global_db_write_cursor.execute(
+                'INSERT INTO custom_assets(identifier, type, notes) VALUES(?, ?, ?)',
+                custom_asset.serialize_for_db(),
+            )
+            self.db.add_asset_identifiers(db_write_cursor, [custom_asset.identifier])
+        return custom_asset.identifier
+
+    def edit_custom_asset(self, custom_asset: CustomAsset) -> None:
+        """
+        Edits a custom asset.
+        May raise:
+        - InputError if editing a custom asset that does not exist or combination
+        of name and type is already present.
+        """
+        self.check_if_custom_asset_exists(custom_asset)
+        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+            write_cursor.execute(
+                'UPDATE assets SET name=? WHERE identifier=?',
+                (custom_asset.name, custom_asset.identifier),
+            )
+            write_cursor.execute(
+                'UPDATE custom_assets SET notes=?, type=? WHERE identifier=?',
+                (custom_asset.notes, custom_asset.custom_asset_type, custom_asset.identifier),
+            )
+            if write_cursor.rowcount == 0:
+                raise InputError(
+                    f'Tried to edit custom asset with identifier {custom_asset.identifier} and name '  # noqa: E501
+                    f'{custom_asset.name} but it was not found',
+                )
+
+    @staticmethod
+    def check_if_custom_asset_exists(custom_asset: CustomAsset) -> None:
+        """
+        This function checks if the custom asset's name & type are not already used in the DB.
+        Raises InputError if match exists.
+        """
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            result = cursor.execute(
+                'SELECT A.identifier, B.name, A.type FROM custom_assets AS A JOIN assets B ON A.identifier = B.identifier WHERE A.type=? AND B.name=?;',  # noqa: E501
+                (custom_asset.custom_asset_type, custom_asset.name),
+            ).fetchone()
+            # if match is found and is being edited with fields left unchanged
+            if result is not None and result != (custom_asset.identifier, custom_asset.name, custom_asset.custom_asset_type):  # noqa: E501
```

Comment:
hmm why do you have the second condition after the `and`? Can you just move the check for all 3 in he sqlite check? The only thing missing from the sqlite check is the identifier. Just do it.